lib/alg/bellman-ford.js

```diff
@@ -0,0 +1,74 @@
+var _ = require("../lodash");
+
+module.exports = bellmanFord;
+
+var DEFAULT_WEIGHT_FUNC = _.constant(1);
+
+function bellmanFord(g, source, weightFn, edgeFn) {
+  return runBellmanFord(g, String(source),
+                     weightFn || DEFAULT_WEIGHT_FUNC,
+                     edgeFn || function(v) { return g.outEdges(v); });
+}
+
+
+function runBellmanFord(g, source, weightFn, edgeFn) {
+  var results = {},
+      didADistanceUpgrade = true,
+      iterations = 0,
+      nodes = g.nodes(),
+      edges = [];
+
+  // Get the edges using the edgeFn provided
+  nodes.forEach(function(v){
+    edgeFn(v).forEach(function(e){
+      if(typeof(e)==="object"){
+        edges.push(e);
+      }
+      else {
+        e.forEach(function(edge) {
+          edges.push(edge);
+        });
+      }
+    });
+  });
+
+  var relaxEdge = function(edge) {
+    var edgeWeight = weightFn(edge);
+    if( results[edge.v].distance + edgeWeight < results[edge.w].distance ){
```

Comment:
The relaxEdge function is called on the items of edges array, which is created in lines 22-33 where I check If the returned value is an object or an array. In other words, I make sure the function's input is always an edge object. If we change the way we are traversing edges, as mentioned above, this will be corrected correspondingly

About traversing the edges backwards, I was about to open an issue on that because I didnt know if it was preferred or not. So, if we call the edge function on a node V, and in a returned edge object, V is the edge.w then I am using the edge as if it was reversed.